exporter/exporterhelper/internal/queue_capacity.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+
+import (
+	"sync/atomic"
+)
+
+type itemsCounter interface {
+	ItemsCount() int
+}
+
+// sizeOfFunc is a function that returns the size of the given element.
+type sizeOfFunc[T any] func(T) uint64
+
+type QueueCapacityLimiter[T any] struct {
+	used *atomic.Uint64
+	cap  uint64
+
+	// sz is a function that returns the size of the given element.
+	// If nil, the size of each element is assumed to be 1.
+	sz sizeOfFunc[T]
+}
+
+func (bcl QueueCapacityLimiter[T]) Capacity() int {
+	return int(bcl.cap)
+}
+
+func (bcl QueueCapacityLimiter[T]) Size() int {
+	return int(bcl.used.Load())
+}
+
+func (bcl QueueCapacityLimiter[T]) claim(el T) bool {
+	size := bcl.sizeOf(el)
+	if bcl.used.Add(size) > bcl.cap {
+		bcl.releaseSize(size)
+		return false
+	}
+	return true
+}
+
+func (bcl QueueCapacityLimiter[T]) release(el T) {
+	bcl.releaseSize(bcl.sizeOf(el))
+}
+
+func (bcl QueueCapacityLimiter[T]) releaseSize(size uint64) {
+	bcl.used.Add(^(size - 1))
+}
+
+func (bcl QueueCapacityLimiter[T]) sizeOf(el T) uint64 {
+	if bcl.sz == nil {
```

Comment:
Don't use nil for request base, force to always provide a func.

Reply:
done